en/companion_computer/pixhawk_rpi.md

```diff
@@ -0,0 +1,259 @@
+# Using the Raspberry Pi with the Pixhawk Controllers
+
+This section describes how to connect the Pixhawk to the Raspberry Pi ("RPi"). Both the software setup and the wire connections between the hardware are presented. 
+
+There are different ways to connect the RPi with the Pixhawk:
+
+1. Via Ethernet port.
+
+2. Serial connection: via `TELEM` port or via `USB` port.
+
+The connection via `TELEM` port can be done with a FTDI Chip USB-to-serial adapter board or directly at the RPi's TX/RX pins. The following steps describes the connections using the TX/RX pins, since no additional board is required.
+
+There are different models of Pixhawk and RPi, therefore, the main concepts are covered and illustrated in order to give the user the necessary knowledge for any other configuration. The first step is the identification of the `TELEM 2` and the `USB` ports in the Pixhawk. The image illustrates the ports in the Pixhawk-6C Model.
+
+<img src="../../assets/companion_computer/pixhawk_rpi/pixhawk_6c.png" width="402" height="452" />
+
+## PX4 setup on Pixhawk
+
+Connect the Pixhawk with the laptop/desktop via `USB` port and check, at the QGroundControl, if the firmware (PX4) is updated. 
+The firmware update can be done in the QGroundControl or following the next steps:
+
+Run in the terminal:
+
+```
+sudo apt -y install gcc-arm-none-eabi make stlink-tools
+```
+
+Clone the last PX4 version and run the script setup:
+
+```
+git clone https://github.com/PX4/PX4-Autopilot.git --recursive 
+PX4-Autopilot/Tools/setup/ubuntu.sh
+```
+
+Build it on the Pixhawk. On Linux, the default name of a USB connection is `/dev/ttyACM0`:
+
+```
+sudo chmod a+rw /dev/ttyACM0
+cd /PX4-Autopilot
+make px4_fmu-v6c_default upload
+```
+
+## Ubuntu setup on RPi
+
+The following steps are required to have the Ubuntu 22.04 installed on the RPi. In the next section, the ROS 2 is going to be installed, so it is important the choice for the correct version of Ubuntu.
+
+Prepare the Ubuntu 22.04's boot SD card following the official tutorial available in the Ubuntu website:
+
+https://ubuntu.com/tutorials/how-to-install-ubuntu-desktop-on-raspberry-pi-4#1-overview
+
+Next, connect the SD card in the RPi. Connect the mouse, keyboard, monitor and connect the RPi on 5v Power Supply (external source/charger).
+
+Turn on the RPi to boot on SD card and install Ubuntu. In the sequence, install raspi-config in the RPi's terminal:
+
+```
+sudo apt update
+sudo apt upgrade
+sudo apt-get install raspi-config 
+```
+
+Open the raspi-config:
+
+```
+sudo raspi-config
+```
+
+Go to the **Interface Option** and then click at **Serial Port**. Select **No** to disable serial login shell and select **Yes** to enable serial interface. Click **finish** and restart the RPi.
+
+In the sequence, in the RPi's terminal:
```

Comment:
Do you mean "follow these commands in RPi;s terminal, in the following order"?

Reply:
> Do you mean "follow these commands in RPi;s terminal, in the following order"?

No ... the "sudo raspi-config" will pop-up a window, like a GUI, in which you need to go and click, select and do everything.

Reply:
OK. What I meant was "what do you mean by in the sequence". I think you mean "do these steps".

Reply:
> OK. What I meant was "what do you mean by in the sequence". I think you mean "do these steps".

Yes, 'do these steps"